### ISSUE TITLE
Lint/UnneededDisable of Metrics/LineLength that isn't unneeded

### ISSUE THREAD

**Issue by jfelchner** [bbatsov/rubocop#4001]
This line (among many others) is throwing an error for me:

```ruby
  LARGE_DATA_STRING_PATTERN = %r{\A([A-Za-z0-9\+\/#]*\={0,2})#([A-Za-z0-9\+\/#]*\={0,2})#([A-Za-z0-9\+\/#]*\={0,2})\z} # rubocop:disable Metrics/LineLength
```

My full `LineLength` config is as follows:

```yaml
Metrics/LineLength:
  Enabled:                       true
  Max:                           90
  AllowHeredoc:                  true
  AllowURI:                      true
  IgnoreCopDirectives:           true
  URISchemes:
    - 'http'
    - 'https'
```

--------

## Expected behavior

No error for `Lint/UnneededDisable`

## Actual behavior

An error for `Lint/UnneededDisable`

## RuboCop version

```
$ rubocop -V
0.47.1 (using Parser 2.3.3.1, running on ruby 2.4.0 x86_64-darwin16)
```

**Comment by mikegee** [comment]
I confirmed the report. Also, changing `IgnoreCopDirectives` to `false` gets the expected behavior (no offense from `Lint/UnneededDisable`). I suspect there's an unexplored edge case.

Sidenote: It would be nice if `IgnoreCopDirectives` were flipped to positive phrasing like `IncludeCopDirectives`, to make this easier to reason about.

**Comment by wkurniawan07** [comment]
As it turns out, it's really easy to reproduce this. Any line containing **both** non-directive `#` **and** directive `#` has the potential to run into this bug.

I'd be happy to submit a fix for this.

**Comment by wkurniawan07** [comment]
@bbatsov 

[The following line](https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/comment_config.rb#L14):
```rb
    COMMENT_DIRECTIVE_REGEXP = Regexp.new(
      ('\A# rubocop : ((?:dis|en)able)\b ' + COPS_PATTERN).gsub(' ', '\s*')
    )
```
Is it necessary for the regexp to have the `\A` assertion? It is one of the culprits behind this bug.
